dpnp/backend/src/dpnpc_memory_adapter.hpp

```diff
@@ -54,19 +55,21 @@ class DPNPC_ptr_adapter final
 public:
     DPNPC_ptr_adapter() = delete;
 
-    DPNPC_ptr_adapter(const void* src_ptr,
+    DPNPC_ptr_adapter(sycl::queue& queue,
+                      const void* src_ptr,
                       const size_t size,
                       bool target_no_sycl = false,
                       bool copy_back_request = false)
     {
+        q = queue;
```

Comment:
We also need to use the queue for memory allocation, copy and free. I mean we need to avoid calls of `dpnp_memory_alloc_c`, `dpnp_memory_memcpy_c` and dpnp_memory_free_c.